app/component/itinerary/navigator/hooks/useRealtimeLegs.js

```diff
@@ -149,7 +172,27 @@ const useRealtimeLegs = (initialLegs, mapRef, relayEnvironment) => {
     return () => clearInterval(interval);
   }, [fetchAndSetRealtimeLegs]);
 
-  return { realTimeLegs, time };
+  let currentIndex = -1;
+
+  if (realTimeLegs) {
+    if (time >= legTime(realTimeLegs[realTimeLegs.length - 1].end)) {
+      currentIndex = realTimeLegs.length - 1;
+    } else {
+      currentIndex = realTimeLegs?.findIndex(
+        l => legTime(l.start) <= time && time <= legTime(l.end),
+      );
+    }
+  }
+
+  return {
+    realTimeLegs,
+    time,
+    origin,
+    firstLeg: realTimeLegs?.[0],
+    lastLeg: realTimeLegs?.[realTimeLegs.length - 1],
```

Comment:
I am not totally against this but why to pass first and last leg as a prop?  Indexing the array is not that expensive or complicated. 

Note that these legs are not consistently used in NaviContainer or NavicardContainer.

Reply:
To centralize the deduction logic instead of having identical pieces of logic in multiple components. Thus, `useRealtimeLegs` becomes responsible for correctness of data from there on out.